### DISCUSSION TITLE
Is there a way to not display ID in the notes

### DISCUSSION THREAD

**Discussion by Yiiipu** [ObsidianToAnki/Obsidian_to_Anki#545] [Q&A]
The ID takes extra spaces and disturb the formatting of the notes. Is there a way to not display them in the notes?

**Comment by Yiiipu** [comment]
ok, seems the only workaround is to use read mode

**Comment by envico801** [comment]
@Yiiipu Indeed, in read mode it would simulate regular markdown. The only option the pluggin offers is to make the comment visible, but that would only make the situation worse.

![image](https://github.com/ObsidianToAnki/Obsidian_to_Anki/assets/132226893/55da2297-196a-4b80-a4e8-96dd5a68547a)

If you really want to remove the ID's you could go so far as to modify the pluggin's `main.js` file, and prevent them from being created in the first place. But I advise against it because these ID's are the plugin's way of detecting when you make a change to the cards or if you want to delete them via obsidian.

**Comment by briana-brady** [comment]
I have a solution that allows me to not see the IDs while typing in Preview Mode or Reading Mode, although I used an additional Obsidian plugin to do so.
I installed the Regex Mark plugin and told it to add a `anki-comment` css class to html comments using the following regex:
```
<!--([\s\S]+?)-->
```
Make sure the `ID Comments: Wrap note IDs in a HTML comment` option is turned on. I don't use any other html comments in my notes, so this only targets Obsidian to Anki IDs.

I then created a CSS snippet that targets the `anki-comment` class:
```css
.cm-line:has(> span.anki-comment) {
    display: none !important;
}
```
This means that I only see IDs in Source Mode, which I never use.

**Reply by AgusssChh** [reply]
Hi! I used your solution for a while and it used to work fine, now it doesn't hide it anymore. But depending on how I toggle the options in the Regex Mark plugin it does sometimes gray out the ID, it just doesn't hide it as it used to.  I don't have other html comments in my notes other than the anki IDs. 
I have tried deleting and re-adding the CSS snippet, deleting and re-adding the regex mark css class, restarting obsidian, updating plugins, but it is still not working. Do you have any idea why it is not hiding the ID?

**Reply by briana-brady** [reply]
It could be that updates to obsidian have changed which css selectors would allow the regex class to work. (`.cm-line:has(> span.anki-comment) ` might need to be changed.)  I would inspect the comment in the developer tools (macOS → cmd+option+i  or Windows → ctrl+shift+i) and see if the css is properly targeting the html comments or if the style is getting overridden by something else. If not, maybe updating the regex mark plugin if there are any available updates. 

I'm no longer using the ObsidianToAnki plugin, but hopefully this points to a helpful direction for debugging.
